src/components/Create/BudgetForm.tsx

```diff
@@ -116,8 +134,23 @@ export default function BudgetForm({
               rules: [{ required: true }],
               extra: null,
             }}
-            value={target.toString()}
-            onValueChange={val => setTarget(val || '0')}
+            target={target}
+            targetSubFee={targetSubFee}
+            onTargetChange={target => {
+              setTarget(target || '0')
```

Comment:
**suggestion (non-blocking)**: we could be a little more strict here (and in other instances below) and use the nullish coalescing operator:

```suggestion
              setTarget(target ?? '0')
```